### ISSUE TITLE
Device join doesn't always trigger

### ISSUE THREAD

**Issue by TheKillerremijn** [zigbeer/zigbee-shepherd#18]
When trying to connect a new device, endDeviceAnnceInd fires, however it does not always start associating

Steps to reproduce:
 - Turn off a device when it is connecting
 - Try to pair another device

**Comment by splitice** [comment]
Yes, this occurs due to the device being joined (both according to ZNP & the device) while Zigbee Shepherd thinks it is not joined.

You can also accomplish this by letting the device fully join and then restarting ZS. Although fixing that behaviour is alot harder.
